spec/custom/index.html

```diff
@@ -584,167 +457,224 @@
+    <li>If <var>localName</var> does not match the <a href="https://www.w3.org/TR/xml/#NT-Name"><code>Name</code></a> production, <a href="https://heycam.github.io/webidl/#dfn-throw">throw</a> an <code>InvalidCharacterError</code> exception.</li>
+
+    <li>If the <a href="https://dom.spec.whatwg.org/#context-object">context object</a> is an <a href="https://dom.spec.whatwg.org/#html-document">HTML document</a>, let <var>localName</var> be <a href="https://dom.spec.whatwg.org/#converted-to-ascii-lowercase">converted to ASCII lowercase</a>.</li>
+
+    <li>Let <var>registry</var> be the <a href="https://dom.spec.whatwg.org/#context-object">context object</a>'s <a>registry</a>.</li>
+
+    <li>
+        <p>If <var>typeExtension</var> is provided:</p>
+
+        <ol>
+            <li>Let <var>definition</var> be the entry in <var>registry</var> with <a href="#dfn-element-definition-type">custom element type</a> <var>typeExtension</var>.</li>
+
+            <li>If <var>definition</var>'s <a href="#dfn-element-definition-local-name">local name</a> is not <var>localName</var>, <a href="https://heycam.github.io/webidl/#dfn-throw">throw</a> a <code>TypeError</code> exception and abort these steps.</li>
+
+            <li>Let <var>interface</var> be the <a href="https://dom.spec.whatwg.org/#concept-element-interface">element interface</a> for <var>localName</var> and the <a href="https://dom.spec.whatwg.org/#html-namespace">HTML namespace</a>.</li>
+
+            <li>Let <var>result</var> be a new <a href="https://dom.spec.whatwg.org/#concept-element">element</a> that implements <var>interface</var>, with no attributes, <a href="https://dom.spec.whatwg.org/#concept-element-namespace">namespace</a> set to the <a href="https://dom.spec.whatwg.org/#html-namespace">HTML namespace</a>, <a href="https://dom.spec.whatwg.org/#concept-element-local-name">local name</a> set to <var>localName</var>, and <a href="https://dom.spec.whatwg.org/#concept-node-document">node document</a> set to the <a href="https://dom.spec.whatwg.org/#context-object">context object</a>.</li>
+
+            <li><a href="https://dom.spec.whatwg.org/#concept-element-attributes-set-value">Set the attribute value</a> for <var>result</var> using <code>is</code> and <var>type</var>.</li>
+        </ol>
+    </li>
+
+    <li>
+        <p>Otherwise, if <var>registry</var> contains an entry with <a href="#dfn-element-definition-local-name">local name</a> <var>localName</var>:</p>
+
+        <ol>
+            <li>Let <var>definition</var> be the entry in <var>registry</var> with <a href="#dfn-element-definition-local-name">local name</a> <var>localName</var>.</li>
+
+            <li>Let <var>C</var> be <var>definition</var>'s <a href="#dfn-element-definition-constructor">constructor</a>.</li>
+
+            <li>Return the result of <a href="https://tc39.github.io/ecma262/#sec-construct">Construct</a>(<var>C</var>), rethrowing any exceptions.</li>
```

Comment:
When speccing this I ended up just doing Construct(C) in createElement. I vaguely recall @annevk wanting to use the upgrading path for this, since it's a DOM method and so it should use the same nanotask infrastructure? @rniwa points out the difference is observable in edge cases where you construct yourself inside your constructor before calling super().